File: api/rest/controllers/api/imaging/application_controller.py
from flask import Blueprint

from api.rest.interfaces.api_response import ApiResponse
from logger import Logger

# Define controller
from services.imaging.application_unnamed_service import ApplicationUnnamedService

application_controller = Blueprint('application', __name__,
                                   url_prefix='/application')

# Declare services & utils
logger = Logger.get_logger("Application Controller")
application_service = ApplicationUnnamedService()


# Define routes
@application_controller.route("/list", methods=['GET'])
def get_application_list():
    """
    Get the list of available applications
    :return:
    """
    try:
        application_list = application_service.get_application_list()
        response = ApiResponse("Application list", application_list, [])
        return response.repr_json(), 200
    except Exception as e:
        logger.error("Failed to get the list of the application", e)
        response = ApiResponse("Application list", None, ["Internal Error. Failed to get the application list"])
        return response.repr_json(), 500

File: client/src/controllers/api/graph/ApplicationController.ts
/**
 * Manage the communication to retrieve Graph Communication
 */
import ProxyAxios from "../../../utils/api/ProxyAxios";
import Status from "../../../interfaces/healthcheck/status";
import Logger from "../../../utils/logging/logger";

export default class ApplicationController {

    private baseRoute: string = "/application"
    private logger: Logger = Logger.getLogger("Application controller");

    /**
     * Build the route
     * @private
     */
    private buildRoute(route: string) {
        return this.baseRoute + route
    }

    /**
     * Get the list of application available on the Neo4j graph database
     */
    public async getApplicationList() : Promise<string[]> {
        try {
            const data = await ProxyAxios.get<string[]>(this.buildRoute("/list"))
            data.checkValidity();
            return data.getData()
        } catch (e) {
            this.logger.error("Failed to get the list of application", e);
            throw new Error("Failed to get the list of application.");
        }
    }
}